### DISCUSSION TITLE
(libreoffice-still) was incorrectly updated to the libreoffice-fresh version

### DISCUSSION THREAD

**Discussion by fraenki** [chocolatey-community/chocolatey-packages#2504] [General]
### Checklist

- [X] I have verified that this is the correct repository, and the package is maintained by the chocolatey-community user.
- [X] I have verified that this is happening in the latest available version of the package.

### Chocolatey Version

2.3.0

### Chocolatey License

None

### Package Version

24.2.4

### Current Behaviour

The package libreoffice-still was updated to version 24.2.4. But version 24.2.4 is a "fresh" release, it's meant for early adopters.

For reference the official release descriptions, which clearly state that 24.2.4 is a "fresh" release:
https://www.libreoffice.org/download/download-libreoffice/

### Expected Behaviour

The correct version for libreoffice-still would be 7.6.7.

### Steps To Reproduce

1. install libreoffice-still
2. version number is 24.2.4, but should be 7.6.7

### Environment

```markdown
- Operating System: all
- PowerShell Version: all
- Shell: all
```

### Chocolatey Log

```shell
none
```

### Anything else?

The incorrect version was introduced in this commit:
https://github.com/chocolatey-community/chocolatey-packages/commit/928f196db54f7f94cd554170e96fc611f97b970a

It looks like the automatic update did something wrong.

**Comment by AdmiringWorm** [comment]
We are reliant on information from the Software Authors using the same mechanism as their built-in updater on what the next version is.

Version 7.6.x went [EOL on June 12th](https://wiki.documentfoundation.org/ReleasePlan/7.6#End_of_Life), and a few days later their built-in updater started providing information about the next version being 24.2.4, and thus our updater pushed this version as `libreoffice-still`.

Before this happened, the latest version of libreoffice-still that this built-in updater suggested was `7.6.6` which is why 7.6.7 was never pushed.

As this appears to be intentional by the software developers, I don't believe there is anything for us to change here. This has also happened in the past as well when the still version went EOL.

We are, however, discussing internally the future of the `-fresh` and `-still` packages for LibreOffice, especially since it seems like they no longer make this notion on their release anymore about it being still or fresh, rather just if it is the current or previous version.

**Comment by AdmiringWorm** [comment]
I forgot to include, there is also confirmation from the Libreoffice Blusky account that the 7.6.x version will be removed from the download page: https://bsky.app/profile/libreoffice.bsky.social/post/3kwexfmfx2x2c

**Comment by fraenki** [comment]
Oh, thanks. So they EOL'd 7.6 but did not officially release a new "still" version. :man_facepalming: So I guess this issue is obsolete...

**Comment by rdiez** [comment]
I just visited the official download page:

https://www.libreoffice.org/download/download-libreoffice/

And version 7.6.7 is offered for download.

**Comment by AdmiringWorm** [comment]
> I just visited the official download page:
> 
> [libreoffice.org/download/download-libreoffice](https://www.libreoffice.org/download/download-libreoffice/)
> 
> And version 7.6.7 is offered for download.

See the bluesky link I provided, they will be removing that download.

It is still EOL.

> Oh, thanks. So they EOL'd 7.6 but did not officially release a new "still" version. 🤦‍♂️ So I guess this issue is obsolete...

Yeah, that is what it looks like happened, and has happened before. But to me, it almost sounds like they won't be keeping a _still_ version anymore (just the previous version available as an option). But we won't know anything about that until we see what happens in late August when 24.8.x is meant to be released, or maybe at the beginning of September.

**Comment by zWhdmB5T** [comment]
> > I just visited the official download page:
> > [libreoffice.org/download/download-libreoffice](https://www.libreoffice.org/download/download-libreoffice/)
> > And version 7.6.7 is offered for download.
> 
> See the bluesky link I provided, they will be removing that download.
> 
> It is still EOL.
> 
> > Oh, thanks. So they EOL'd 7.6 but did not officially release a new "still" version. 🤦‍♂️ So I guess this issue is obsolete...
> 
> Yeah, that is what it looks like happened, and has happened before. But to me, it almost sounds like they won't be keeping a _still_ version anymore (just the previous version available as an option). But we won't know anything about that until we see what happens in late August when 24.8.x is meant to be released, or maybe at the beginning of September.

@AdmiringWorm @TheCakeIsNaOH @pauby 

- I suppose that they won't remove the actual possibility to download — all downloads are here [Index of /libreoffice/old](https://downloadarchive.documentfoundation.org/libreoffice/old/) (from some very early version up to the current one).
- I suppose that they will, indeed, remove the references from [Download LibreOffice](https://www.libreoffice.org/download/download-libreoffice/) and similar sites (just giving "hidden" references for the previous version (aka Still)…).
- I had a conversation with members of the infrastructure team of The Document Foundation: they want to push Fresh, i.e. the current version, over Still, i.e. the previous version, generally.
- They discuss just using "current version" and "previous version" over "Fresh" and "Still" — and currently, they actually do so at the very most.
- They discuss that the current version (aka Fresh) shall become the regular and main version — and the previous version (aka Still) shall just be offered as it (just for the users that really want this — they say that most people want the current version).
- Furthermore, they tell that 3rd-party projects (like Chocolatey) should better not fully rely on the update-checker, because it is not intended to do so for "some" reason (please, ask them for full details).

**Comment by rdiez** [comment]
> * Furthermore, they tell that 3rd-party projects (like Chocolatey) should better not fully rely on the update-checker, because it is not intended to do so for "some" reason (please, ask them for full details).

We had a discussion in the past about the The Document Foundation's behaviour in this respect. For example, search for "We have talked about this a few times" here:

https://community.chocolatey.org/packages/libreoffice-still#comment-5243891998

My opinion in short: The "still" Chocolatey package should mirror what the official download page for humans shows.

If you set up an automatic e-mail notification when that page changes, it shouldn't mean too much manual work. After all, new releases do not come every day, and it looks like it's just a matter of updating a download link.

About the automatic notification, there probably are several online services for that, or you can use Firefox's "Live Bookmark", or completely automate it with some HTML scraping tool. I have scripted such a task in the past by converting the web page to text with w3m, and checking whether the text has changed with "git diff --unified=0 --word-diff --no-index --exit-code old.txt new.txt" (no Git repository necessary). If so, then send a notification e-mail. Use cron to run the script daily in the background.

**Comment by zWhdmB5T** [comment]
@AdmiringWorm @TheCakeIsNaOH @pauby 
@rdiez 

[(libreoffice-still) was incorrectly updated to the libreoffice-fresh version · Issue #2496 · issuecomment-2208491099](https://github.com/chocolatey-community/chocolatey-packages/issues/2496#issuecomment-2208491099) still holds — but to offer more details:

- They explicitly point to [ReleasePlan](https://wiki.documentfoundation.org/ReleasePlan). Aside from emergency updates (sometimes, at time) you get rather precise info about releases, versions, updates.
- Their update checker is always intentionally(!) behind the actual current releases. This is supposed to be at least one week up to several(?) weeks.
- This means that you may see new releases at [Download LibreOffice](https://www.libreoffice.org/download/download-libreoffice/), but the update checker is not reporting them, by design(!).
- So, they do offer new releases available to download — but they don't push you to update.
- This is due to have them the chance to observe (so far) unreported bugs and fix them… before the majority of users update.
- They suggest some alternative method for 3rd parties to check for updates and/or actually update: `rsync rsync://download.documentfoundation.org/tdf-pub/libreoffice/stable/` and for RC and pre-releases `rsync rsync://download.documentfoundation.org/tdf-pub/libreoffice/testing/`.

**Comment by rdiez** [comment]
I checked the LibreOffice download page today, and the "still" version is no longer prominently displayed.

The LibreOffice foundation seems to have buckled and now obviously tries to get people to pay ("Business users: click here"), or be happy with the newest and probably less stable version.

I would describe this change as step in the way of what is nowadays known as [enshittification](https://en.wikipedia.org/wiki/Enshittification). The first step was the "community" labeling.

If you scroll further down the download page, you will find section titled "Available Versions", where the "still" concept apparently still applies:

LibreOffice is available in the following released versions:
7.6.7
24.2.4

However, the page does not offer a download link for the 7.6.7 version: if you click through, only the binaries for 24.2.4 are actually downloadable. I cannot tell whether this error is on purpose, but it wouldn't surprise me.

In my opinion, Chocolate package libreoffice-still has now 2 options:
1) Remove the package from the Chocolate repository, or at least mark it as obsolete.
2) Still try to track the 7.6.7 version, as long as such an older version is still listed in the download page under "Available Versions", and as long as the Document Foundation does not actively discourage using such a version.

Otherwise, maintaining a "still" package which is the same as the "fresh" one does not really make sense.

**Comment by pauby** [comment]
I've reopened this and converted it to a discussion as it's now more of a 'what should we do' than a 'this package is broken'.

**Comment by TheCakeIsNaOH** [comment]
My suggestion would be to deprecate both the `still` and `fresh` packages back toward the `libreoffice` package. That way, everyone using either package continues to get updates.

**Comment by zWhdmB5T** [comment]
[Download LibreOffice](https://www.libreoffice.org/download/download-libreoffice/) now announces 24.2.5 and 24.2.4 as **released** versions and 24.8.0 as **prerelease** versions. There is no announcement of **previous** versions (i.e. 7.6.7 in this case), any more. However, [Release Notes](https://www.libreoffice.org/download/release-notes/) still announces **24.2.5 (2024-07-11) - Latest Release** and **7.6.7 (2024-05-10) - Previous Release**. Let's see when they will remove _previous_ release there maybe.

Though, the update checkers — both for Fresh and Still — still show up 24.2.4 — not 24.2.5. But [24.4.5.2](https://downloadarchive.documentfoundation.org/libreoffice/old/24.2.5.2/) shows date 11-Jul-2024. So, the lag of the update checker is currently about 1,5 weeks. Also compare [Release Plan 24.2](https://wiki.documentfoundation.org/ReleasePlan/24.2)
